js/Schemas/pluginDefinitionSchema.json

```diff
@@ -0,0 +1,345 @@
+{
+   "$schema": "http://json-schema.org/schema#",
+   "type": "object",
+   "if": {
+      "properties": {
+         "pluginType": {
+            "const": "library"
+         }
+      }
+   },
+   "then": {
+      "required": [
+         "libraryName",
+         "libraryVersion"
+      ]
+   },
+   "else": {
+      "if": {
+         "properties": {
+            "pluginType": {
+               "const": "nodeAuthentication"
+            }
+         }
+      },
+      "then": {
+         "required": [
+            "authenticationCategory",
+            "filename"
+         ]
+      },
+      "else": {
+         "if": {
+            "properties": {
+               "pluginType": {
+                  "const": "desktop"
+               }
+            }
+         },
+         "then": {
+            "properties": {
+               "webContent": {
+                  "if": {
+                     "additionalProperties": false
+                  },
+                  "then": {},
+                  "else": {
+                     "webContentSchema",
```

Comment:
can you describe what webContentSchema is doing here? I can't. It's definitely invalid JSON so I'm thinking this should just be removed.

Reply:
I am guessing the intent was to refer to one schema from another. I think this page describes the correct way to do it: https://json-schema.org/understanding-json-schema/structuring.html#extending